src/merkle.rs

```diff
@@ -10,7 +10,11 @@ pub struct Node<'a> {
 
 impl<'a> Node<'a> {
     pub fn new(value: Vec<&'a str>) -> Self {
```

Comment:
I would probably just use `String` instead of `&str` here to avoid carrying around lifetimes.